src/EventuallyQueue.js

```diff
@@ -348,20 +351,36 @@ const EventuallyQueue = {
     polling = flag;
   },
 
+  /**
+   * Return true if processing the queue.
+   *
+   * @function isProcessing
+   * @name Parse.EventuallyQueue.isProcessing
+   * @returns {boolean}
+   * @static
+   */
+  isProcessing(): boolean {
+    return processing;
+  },
+
+  _setProcessing(flag: boolean) {
+    processing = flag;
+  },
+
   process: {
     create(ObjectType, queueObject) {
       const object = new ObjectType();
       return EventuallyQueue.sendQueueCallback(object, queueObject);
     },
     async byId(ObjectType, queueObject) {
-      const { sessionToken } = queueObject.serverOptions;
+      const { sessionToken = '' } = queueObject.serverOptions;
       const query = new ParseQuery(ObjectType);
       query.equalTo('objectId', queueObject.id);
       const results = await query.find({ sessionToken });
       return EventuallyQueue.sendQueueCallback(results[0], queueObject);
     },
     async byHash(ObjectType, queueObject) {
-      const { sessionToken } = queueObject.serverOptions;
+      const { sessionToken = '' } = queueObject.serverOptions;
```

Comment:
Why should it execute a query if the session token is an empty string? Wouldn't that query always fail?

Reply:
The opposite, the query always runs regardless if the user's session token is invalid.

RESTController.js
```
        const userController = CoreManager.getUserController();
        if (options && typeof options.sessionToken === 'string') {
          return Promise.resolve(options.sessionToken);
        } else if (userController) {
          return userController.currentUserAsync().then(user => {
            if (user) {
              return Promise.resolve(user.getSessionToken());
            }
            return Promise.resolve(null);
          });
        }
        return Promise.resolve(null);
```

Reply:
What's the point of sending an empty token to the server? Could such a query ever succeed?

Reply:
The query does succeed unless there is a permission issue

Reply:
You mean for example if there is a public write permission the query would succeed?

Reply:
Yes

Reply:
Got it, and why send an empt string instead of not setting the sessionToken? An empty string could technically be considered a token of zero length, i.e. an invalid token. Even if Parse Server don't treat it this way now, it could in the future.